.github/workflows/gcp-dsh-testing.yaml

```diff
@@ -0,0 +1,208 @@
+name: "[GCP] DevZero self-hosted deployment"
+run-name: '[GCP] self-hosted test run'
+
+on:
+  push:
+    branches:
+      - garvit/gcp-tf
+  workflow_dispatch:
+
+jobs:
+  setup-and-test:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: 'Authenticate to Google Cloud'
+        id: 'auth'
+        uses: 'google-github-actions/auth@v0.4.0'
+        with:
+          workload_identity_provider: 'projects/354385559261/locations/global/workloadIdentityPools/dz-ga-dsh-pool/providers/dz-ga-dsh-provider'
+          service_account: 'dsh-testing-github-actions@devzero-kubernetes-sandbox.iam.gserviceaccount.com'
```

Comment:
Can we not call things `sandbox`? This is not a sandbox, this is self hosted testing